synapse/storage/databases/main/events_bg_updates.py

```diff
@@ -1140,32 +1142,49 @@ def _event_arbitrary_relations_txn(txn: LoggingTransaction) -> int:
                 if not relates_to or not isinstance(relates_to, dict):
                     continue
 
-                # The only expected relation type would be from threads, but
-                # there could be other unknown ones.
+                # If the relation type or parent event ID is not a string a
+                # string, skip it.
+                #
+                # Do not consider relation types that have existed for a long time,
+                # only include the new thread relation and any unknown relations.
```

Comment:
```suggestion
                # Do not consider relation types that have existed for a long time,
                # since they will already be listed in the `event_relations` table.
```